crates/matrix-sdk-ui/src/timeline/inner.rs

```diff
@@ -496,25 +496,20 @@ impl<P: RoomDataProvider> TimelineInner<P> {
             _ => None,
         };
 
-        // Look for the local event by the transaction ID or event ID.
-        let result = rfind_event_item(&state.items, |it| {
-            it.transaction_id() == Some(txn_id)
-                || new_event_id.is_some() && it.event_id() == new_event_id
-        });
-
-        let Some((idx, item)) = result else {
-            // Event isn't found at all.
-            warn!("Timeline item not found, can't add event ID");
-            return;
-        };
-
-        let Some(local_item) = item.as_local() else {
+        // The local echoes are always at the end of the timeline, we must first make
+        // sure the remote echo hasn't showed up yet.
+        // TODO this is expensive to use `rfind_event_item` in that case!
```

Comment:
Do you mean it's more expensive here than elsewhere? I think it should be fine, we just need to eventually add some ways to control the maximum size of the timeline (there's really no reason to hold many hundrets or thousands of events in memory).

Reply:
Now we're doing two `rfind_event_item` for each `update_event_send_state` while we were doing one before, and the one for the remote_echo is almost guaranteed to not succeed most of the time, thus iterating over all the elements. We can figure something out later :-)